commonknowledge/meep-intelligence-hub
# Display hub layers on the hub map

Very basic display of events on a hub map.

## Description

- Add public sharing option to external data sources
- Handle tile-server permissions for anonymous users
- Render hub layers on the hub map (using hardcoded TCC event icon for now)

## Motivation and Context

## How Can It Be Tested?

Add data layers to the root page of the hub, in Wagtail:

<img width="626" alt="Screenshot 2024-05-29 at 16 39 42" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/2cbc3394-9646-4b36-a773-ce96c7dd1b47">
    
Then ensure `can_display_points_publicly` and `can_display_details_publicly` are set to True in the database for the relevant external data sources.

You must also add the custom domain to the `CORS_ALLOWED_ORIGINS` django env, to allow incoming API requests. The env should be strings delimited by comma, e.g. `https://prototype.mapped.commonknowledge.coop,https://united2024.com`

Now go to the hub public url `/map` path and you should see these events show up

## How Will This Be Deployed?

Add any existing custom domains to `CORS_ALLOWED_ORIGINS`

## Screenshots (if appropriate):

<img width="662" alt="Screenshot 2024-05-29 at 16 42 40" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/79f9ba71-b85d-4bae-970c-46ad78ece057">

## Types of changes
